### PR TITLE
Fixes #7466: Templates and Files \"by path\" don't trigger technique library reload

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/7466

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/cfclerk/services/impl/GitTechniqueReader.scala [hunk 1]
```diff
@@ -748,14 +745,29 @@ class GitTechniqueReader(
...
    */
-  private[this] def getTechniquePath(techniqueInfos:TechniquesInfo) : Set[TechniquePath] = {
-   var set = scala.collection.mutable.Set[TechniquePath]()
-   techniqueInfos.rootCategory.packageIds.foreach { p => set += TechniquePath( "/" + p.toString) }
-   techniqueInfos.subCategories.foreach { case (id,cat) =>
-     val path = id.toString
-     cat.packageIds.foreach { p => set += TechniquePath(path + "/" + p.toString) }
+  private[this] def getTechniquePath(techniqueInfos:TechniquesInfo) : Set[(TechniquePath, TechniqueId)] = {
+    var set = scala.collection.mutable.Set[(TechniquePath, TechniqueId)]()
+    techniqueInfos.rootCategory.packageIds.foreach { id =>
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/cfclerk/services/impl/GitTechniqueReader.scala]: Next refactoring, rename packageIds :)

File: rudder-core/src/main/scala/com/normation/cfclerk/services/impl/GitTechniqueReader.scala [hunk 2]
```diff
@@ -748,14 +745,29 @@ class GitTechniqueReader(
...
+    techniqueInfos.subCategories.foreach { case (id,cat) =>
+      val path = id.toString
+      cat.packageIds.foreach { t => set += ((TechniquePath(path + "/" + t.toString), t)) }
+    }
+    //also add template "by path"
+    val techniques = techniqueInfos.techniques.flatMap { case(_, set) => set.map { case(_, t) => t } }
+    techniques.foreach { t =>
+      val byPath = t.templates.collect { case TechniqueTemplate(id@TechniqueResourceIdByPath(_,_),_,_) => id } ++
+                   t.files.collect { case TechniqueFile(id@TechniqueResourceIdByPath(_,_),_,_) => id }
+      byPath.foreach { x =>
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/cfclerk/services/impl/GitTechniqueReader.scala]: Not at ease with the x variable here ... but i can understand it's hard to name it with a understandable name :)
resource ? 
techniqueResource? 
x ? :) 

Maybe it miss a type or a better name for byPath variable so we understand what we handle here

**Comment by fanf** [comment]
PR updated

**Comment by VinceMacBuche** [comment]
Great \o/ Merging

**Pull request merged**